node/src/ledger/ledger_service.rs

```diff
@@ -837,6 +855,54 @@ impl<T: LedgerService> RpcLedgerService for T {
     }
 }
 
+/// Save staged ledger and block to file, when the application fail.
+/// So we can easily reproduce the application both in Rust and OCaml, to compare them.
+/// - https://github.com/openmina/openmina/blob/8e68037aafddd43842a54c8439baeafee4c6e1eb/ledger/src/staged_ledger/staged_ledger.rs#L5959
+/// - TODO: Find OCaml link, I remember having the same test in OCaml but I can't find where
+fn dump_application_to_file(
+    staged_ledger: &StagedLedger,
+    block: ArcBlockWithHash,
+    pred_block: ArcBlockWithHash,
+) -> std::io::Result<String> {
+    use mina_p2p_messages::{
+        binprot,
+        binprot::macros::{BinProtRead, BinProtWrite},
+    };
+
+    #[derive(BinProtRead, BinProtWrite)]
+    struct ApplyContext {
+        accounts: Vec<v2::MinaBaseAccountBinableArgStableV2>,
+        scan_state: v2::TransactionSnarkScanStateStableV2,
+        pending_coinbase: v2::MinaBasePendingCoinbaseStableV2,
+        pred_block: v2::MinaBlockBlockStableV2,
+        blocks: Vec<v2::MinaBlockBlockStableV2>,
+    }
+
+    let cs = &block.block.header.protocol_state.body.consensus_state;
+    let block_height = cs.blockchain_length.as_u32();
+
+    let apply_context = ApplyContext {
+        accounts: staged_ledger
+            .ledger()
+            .to_list()
+            .iter()
+            .map(v2::MinaBaseAccountBinableArgStableV2::from)
+            .collect::<Vec<_>>(),
+        scan_state: staged_ledger.scan_state().into(),
+        pending_coinbase: staged_ledger.pending_coinbase_collection().into(),
+        pred_block: (&*pred_block.block).clone(),
```

Comment:
Very minor thing but I think `&` is extra here and everywhere where clone is done right after, since clone takes re`&self`.

Reply:
@binier You're right. Thanks